sklearn/tree/tree.py

```diff
@@ -971,6 +971,10 @@ def predict_log_proba(self, X):
 
             return proba
 
+    @property
```

Comment:
wouldn't it be easier to implement a setter and getter? but don't have a strong opinion.

Reply:
Or we can have `classes_` set in the base class only if it's a classifier, and then have an attribute which returns `[None] * self.n_outputs_` in the regression tree.

Reply:
That's what master is doing, but I don't think it makes any sense for a regressor to have `classes_`

Reply:
I agree with @NicolasHug. It is surprising to have a `classes_` attribute in a regressor.

Reply:
What I meant to say, was that instead of setting `classes_` in the base class, we can have an attribute in the regression tree which returns that same value (for now) AND raises the deprecation warning. Later, we can just remove that attribute w/o changing any other part of the code, and we don't have to introduce the `@property` and the new `_classes` attribute in the base class.

Reply:
you don't agree with this one @NicolasHug ?

Reply:
I just don't see how that is simpler?

Reply:
Or to be perfectly honest: as long as the current solution works and isn't totally wrong, I'm too lazy to consider another solution unless it's obviously much better.

Reply:
having `classes_` and `_classes` is just confusing, and is adding a private attribute to the object where it doesn't really need to.

Reply:
OK.

I don't understand your suggestion though: how do you make sure that a deprecation warning is raised when `reg.classes_` is accessed, and how do you make sure that ``reg.classes_` throws an error in 0.24?

Reply:
Are you suggesting to use `if isclassifier()` every single time `classes_` is used in the current code? I'd rather not do that

Reply:
Ok, let me double-check, I may be missing something.